src/Acquisition/__init__.py

```diff
@@ -1,8 +1,833 @@
+from __future__ import absolute_import, print_function
+
+# pylint:disable=W0212,R0911,R0912
+
+import os
+import operator
+import sys
+import types
+
+import ExtensionClass
+
 from zope.interface import classImplements
 
-from _Acquisition import *
-from interfaces import IAcquirer
-from interfaces import IAcquisitionWrapper
+from .interfaces import IAcquirer
+from .interfaces import IAcquisitionWrapper
+
+class Acquired(object):
+    "Marker for explicit acquisition"
+
+
+_NOT_FOUND = object()  # marker
+
+###
+# Helper functions
+###
+
+
+def _has__of__(obj):
+    """Check whether an object has an __of__ method for returning itself
+    in the context of a container."""
+    # Note specifically this is a type check, not duck typing, or
+    # we get into cycles
+    return isinstance(obj, ExtensionClass.Base)
```

Comment:
Good point about the C implementation. In pure Python wasn't clear to me whether it would be likely to get in the state where `isinstance(o,Base) == True` yet `getattr(o,'__of__')` would raise---that takes some special circumstances (does Zope somewhere mixin `Base` as a base-class of PersistentMapping post definition?)

I'll try to set up a testcase and make that change.